### PR TITLE
CORTX-33647: Confstore Consul backend should query consul while get.

### PR DESCRIPTION
**Pull request by rahul-tripathi-git**
Signed-off-by: Rahul Tripathi <rahul.tripathi@seagate.com>

# Problem Statement
- Conf.get for consul backend is first checking the built-in (cached) key_list, if the key is not present in cached key list, it is simply returning None. Now, if we load couple of consul conf indexes from two separate python kernels.  Then set a key:value from 1st index and try to fetch the value for the same key from 2nd index(which is on a separate python kernel), it wont be able to find the key in cached key_list and hence return None.
- This is as per current logic and it was done to reduce calls to consul KV over the network.

# Design
- The Logic for checking cached key_list is removed and now every get call is directly going to consul KV server to get the value.

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing
- Are test cases updated along with code changes due to Enhancements/Bugs [Y/N]: NA
- Confirm that new test cases are added to regression and sanity plan files and relevant feature plan files [Y/N]: NA
- Confirm that Test Cases are added for new features added [Y/N]: NA
- Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: NA
- Confirm Testing was performed with installed RPM/K8s deployment [Y/N]:  K8s deployment will happen with this PR via CICD

# Review Checklist 
####  Before posting the PR please ensure:
- [*] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [ ] New package/s added to setup.py?
- [ ] Jira is updated
- [ ] Check if the description is clear and explained. 
- [ ] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified.

# KVstore/Confstore feature (changes/fixes) checklist
#### Confirm changes are made for:
- [ ] ConfStore
- [ ] KVStore
- [ ] ConfCli
- [ ] Test cases added for all above 3
- [ ] changes done in all the KVpayloads (ConsulKvPayload, IniKvPayload, KvPayload)

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

**Review by kailashyadav11 — APPROVED** [review]

File: py-utils/src/utils/kv_store/kv_store_collection.py [hunk 1]
```diff
@@ -433,8 +433,6 @@ def __init__(self, consul: Consul, store_path: str = '', delim: str = '>'):
     @ExponentialBackoff(exception=(ConsulException, HTTPError, RequestException), tries=4)
     def get(self, key: str, *args, **kwargs) -> str:
         """Get value for consul key."""
-        if not key in self._keys:
-            return None
```
Comment (rahul-tripathi-git) [inline py-utils/src/utils/kv_store/kv_store_collection.py:437]: it was done to reduce network calls to consul KV. 

Consul.get first checks the cached key_list unique to loaded consul index, if the key is not present, it assumes that key:val is not present in consul kv and avoides a network call.

The above logic seems not working when 2 separate python kernels are getting involved as K:V set by first index wont add the key in chached key_list of 2nd index (NOTE, both the indexes are in separate python kernels)

Comment (rahul-tripathi-git) [reply py-utils/src/utils/kv_store/kv_store_collection.py:437]: Hi @ujjwalpl 
Please have a look at this change. Kindly review.

**Review by cdeshmukh — APPROVED** [review]

**Review by sachitanands — APPROVED** [review]
looks good

**Comment by rahul-tripathi-git** [comment]
retest this please.

**Comment by cdeshmukh** [comment]
image
1N deployment was also successfull

**Pull request merged**